pkg/dbmate/db.go

```diff
@@ -215,7 +236,60 @@ func (db *DB) DumpSchema() error {
 	}
 
 	// write schema to file
-	return os.WriteFile(db.SchemaFile, schema, 0o644)
+	if err = os.WriteFile(db.SchemaFile, schema, 0o644); err != nil {
+		return err
+	}
+
+	// in prune mode we delete all migrations from disk after dump
+	if db.Prune {
```

Comment:
To implement squashing for migrations after dump we simply clean migrations folder, as that dump already represents latest state